db/user_test.go

```diff
@@ -14,6 +15,46 @@ type SyncNewUsersTestCase struct {
     sendEmails bool
 }
 
+func (this *DBTests) DBTestResolveUsers(test *testing.T) {
+    defer ResetForTesting();
+
+    testCases := []struct{input []string; expectedOutput []string; course *model.Course; addUsers []*model.User; removeUsers []string} {
+        {[]string{""}, make([]string, 0, 1), nil, nil, make([]string, 0, 1)},
+        {[]string{"b@test.com", "a@test.com", "c@test.com"}, []string{"a@test.com", "b@test.com", "c@test.com"}, nil, nil, make([]string, 0, 1)},
+        {[]string{"admin"}, []string{"admin@test.com"}, MustGetCourse(TEST_COURSE_ID), nil, make([]string, 0, 1)},
+        {[]string{"*"}, []string{"admin@test.com", "grader@test.com", "other@test.com", "owner@test.com", "student@test.com"},
+            MustGetCourse(TEST_COURSE_ID), nil, make([]string, 0, 1)},
+        {[]string{"other", "*", "grader@test.com", "zoinks@test.com", "ZoinKS@teSt.Com"}, []string{"admin@test.com", "grader@test.com",
```

Comment:
Make these test cases more readable.
Put newlines between test cases.
Try not to break items, but break between items.

Reply:
Would you rather an incredibly lone single line for these cases?